src/createTwitchVideoCompiler/youtubeUploader.js

```diff
@@ -0,0 +1,192 @@
+// @flow
+
+import fs from 'fs';
+import readline from 'readline';
+import { google } from 'googleapis';
+
+export default async function youtubeUploader(
+  clientFilePath: string,
+  properties: {
+    language: string,
+    description: string,
+    tags: string,
+    title: string,
+    privacy: string,
+    videoPath: string,
+  },
+): Promise<void> {
+  return new Promise((resolve, reject) => {
+    const { OAuth2 } = google.auth;
+    const SCOPES = ['https://www.googleapis.com/auth/youtube.force-ssl'];
+    // $FlowFixMe
+    const TOKEN_DIR = `${process.env.HOME ||
+      // $FlowFixMe
+      process.env.HOMEPATH ||
+      // $FlowFixMe
+      process.env.USERPROFILE}/.credentials/`;
+    const TOKEN_PATH = `${TOKEN_DIR}twitch-parser.json`;
+
+    function storeToken(token) {
+      try {
+        fs.mkdirSync(TOKEN_DIR);
+      } catch (err) {
+        if (err.code !== 'EEXIST') {
+          throw err;
+        }
+      }
+      fs.writeFile(TOKEN_PATH, JSON.stringify(token));
+      console.log(`Token stored to ${TOKEN_PATH}`);
+    }
+
+    function getNewToken(oauth2Client, requestData, callback) {
+      const authUrl = oauth2Client.generateAuthUrl({
+        access_type: 'offline',
+        scope: SCOPES,
+      });
+
+      console.log('Authorize this app by visiting this url: ', authUrl);
+
+      const rl = readline.createInterface({
+        input: process.stdin,
+        output: process.stdout,
+      });
+
+      rl.question('Enter the code from that page here: ', (code) => {
+        rl.close();
+
+        oauth2Client.getToken(code, (err, token) => {
+          const client = oauth2Client;
+
+          if (err) {
+            console.log('Error while trying to retrieve access token', err);
+            return;
+          }
+
+          client.credentials = token;
+
+          storeToken(token);
+
+          callback(client, requestData);
+        });
+      });
+    }
+
+    function removeEmptyParameters(params) {
```

Comment:
I know this is some copy&paste code, but in the future we might want to move all these functions outside of this file and refactor them.